algosdk/future/atomic_transaction_composer.py

```diff
@@ -0,0 +1,512 @@
+        committed to a block. An error will be thrown if submission fails.
+        The composer's status must be SUBMITTED or lower before calling this method.
+        If submission is successful, this composer's status will update to SUBMITTED.
+
+        Note: a group can only be submitted again if it fails.
+
+        Returns:
+            list[Transaction]: list of submitted transactions
+        """
+        if self.status <= AtomicTransactionComposerStatus.SUBMITTED:
+            self.gather_signatures()
+        else:
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposerStatus must be submitted or lower to submit a group"
+            )
+
+        client.send_transactions(self.signed_txns)
+        self.status = AtomicTransactionComposerStatus.SUBMITTED
+        return self.tx_ids
+
+    def execute(self, client, wait_rounds):
+        """
+        Send the transaction group to the network and wait until it's committed
+        to a block. An error will be thrown if submission or execution fails.
+        The composer's status must be SUBMITTED or lower before calling this method,
+        since execution is only allowed once. If submission is successful,
+        this composer's status will update to SUBMITTED.
+        If the execution is also successful, this composer's status will update to COMMITTED.
+
+        Note: a group can only be submitted again if it fails.
+
+        Returns:
+            AtomicTransactionResponse: Object with confirmed round for this transaction,
+                a list of txIDs of the submitted transactions, and an array of
+                results for each method call transaction in this group. If a
+                method has no return value (void), then the method results array
+                will contain None for that method's return value.
+        """
+        if self.status > AtomicTransactionComposerStatus.SUBMITTED:
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposerStatus must be submitted or lower to execute a group"
+            )
+
+        self.submit(client)
+        transaction.wait_for_confirmation(client, self.tx_ids[0], wait_rounds)
+        self.status = AtomicTransactionComposerStatus.COMMITTED
+
+        confirmed_round = -1
+        method_results = []
+
+        for i, tx_id in enumerate(self.tx_ids):
+            if i not in self.method_dict:
+                continue
+            # Return is void
+            if self.method_dict[i].returns is None:
+                method_results.append(
+                    ABIResult(
+                        tx_id=tx_id,
+                        raw_value=None,
+                        return_value=None,
+                        decode_error=None,
+                    )
+                )
+                continue
+
+            # Parse log for ABI method return value
+            resp = client.pending_transaction_info(tx_id)
```

Comment:
It would better if you can expand the `try` block to include this call too, since it might error due to network issues.